### PR TITLE
PROD-755 BuddyBoss component pages are rendered blank on Platform + Multisite + TranslatePress - Fixed

### LINKED ISSUE #2835: BuddyBoss component pages are rendered blank on Platform + Multisite + TranslatePress setup
**Describe the bug**
BuddyBoss Pages are rendered blank on Platform + Multisite + TranslatePress setup. Please see this video: https://www.loom.com/share/15085bb212ce4fedba9dcd1741c19323

**To Reproduce**
1. Create a Multisite environment
2. Install BuddyBoss Theme, Platform and TranslatePress
3. In TranslatePress, create another language aside from default language
4. In front end, check BuddyBoss pages (activity page for instance ) and try to switch language.
5. See error. Page will be blank

**Environment**
Local, client's live site, dev site and anywhere

**Support ticket links**
https://secure.helpscout.net/conversation/1601359763/158755?folderId=3709082

**Jira issue** : [PROD-755]

[PROD-755]: https://buddyboss.atlassian.net/browse/PROD-755?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by mahedihasannoman**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2835  .

### How to test the changes in this Pull Request:

1. Create a Multisite environment
2. Install BuddyBoss Theme, Platform and TranslatePress
3. In TranslatePress, create another language aside from default language
4. In front end, check BuddyBoss pages (activity page for instance ) and try to switch language.
5. See error is not happening anymore

### Proof Screenshots or Video

https://www.loom.com/share/ed7b21c45676430fa070a74208157588

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by dipskakadiya — APPROVED** [review]

**Review by dipskakadiya — APPROVED** [review]

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php [hunk 1]
```diff
@@ -154,6 +154,32 @@ function bp_core_instructor_role_post_exclude( $exclude_posts ) {
...
 	}
+
+	if ( in_array( 'translatepress-multilingual/index.php', $bp_plugins, true ) && is_multisite() ) {
+
+		/**
+		 * Exclude translatepress lang slug from URL path for only multisite
+		 * 
+		 * @since BuddyBoss [BBVERSION]
+		 *
+		 * @param string $url URL with Language code
```
Comment (KartikSuthar) [inline src/bp-core/compatibility/bp-incompatible-plugins-helper.php:165]:
```suggestion
		 * @param string $url URL with Language code.
```

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php [hunk 2]
```diff
@@ -154,6 +154,32 @@ function bp_core_instructor_role_post_exclude( $exclude_posts ) {
...
+		 * 
+		 * @return string $url URL without language code
```
Comment (KartikSuthar) [inline src/bp-core/compatibility/bp-incompatible-plugins-helper.php:167]:
```suggestion
		 * @return string $url URL without language code.
```

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php [hunk 3]
```diff
@@ -154,6 +154,32 @@ function bp_core_instructor_role_post_exclude( $exclude_posts ) {
...
+			if( ! is_admin() ) {
+				$tp_settings = get_option( 'trp_settings' );
+				if ( $tp_settings && isset( $tp_settings['url-slugs'] ) && !empty( $tp_settings['url-slugs'] ) ) {
+					foreach ( $tp_settings['url-slugs'] as $lang_key => $lang_slug ) {
+					$url = str_replace( $lang_slug . '/', '', $url );
+					}
+				}
+			}
+			return $url;
+		}
```
Comment (KartikSuthar) [inline src/bp-core/compatibility/bp-incompatible-plugins-helper.php:179]:
```suggestion
		function bb_core_multisite_multilang_translatepress_exclude_lang_slug( $url ) {
			if ( ! is_admin() ) {
				$tp_settings = get_option( 'trp_settings' );
				if ( $tp_settings && isset( $tp_settings['url-slugs'] ) && ! empty( $tp_settings['url-slugs'] ) ) {
					foreach ( $tp_settings['url-slugs'] as $lang_key => $lang_slug ) {
						$url = str_replace( $lang_slug . '/', '', $url );
					}
				}
			}

			return $url;
		}
```

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

File: src/bp-core/compatibility/bp-incompatible-plugins-helper.php (hunk 3 above)
Comment (mahedihasannoman) [reply src/bp-core/compatibility/bp-incompatible-plugins-helper.php]: Done

**Comment by KartikSuthar** [comment]
Its duplicate PR: https://github.com/buddyboss/buddyboss-platform/pull/2950/files

**Pull request closed without merging**